### PR TITLE
[RFC] Special interfaces cleanup

### PR DESCRIPTION
**Pull request by ghost**
refs #1646

From the commit message:

```
    This removes the ability to start nvim via the following aliases in favor of just
    using their command line arguments:

    ex        vim -e
    exim      vim -E
    view      vim -R
    gvim      vim -g
    gex       vim -eg
    gview     vim -Rg
    rvim      vim -Z
    rview     vim -RZ
    rgvim     vim -gZ
    rgview    vim -RgZ

    This also removes Vi mode (-v) for consistency. From ':help -v':

        -v  Start Ex in Vi mode.  Only makes a difference when the
        executable is called "ex" or "gvim".  For gvim the GUI is not
        started if possible.
```

I didn't completely remove all refs to these interfaces in the non-english man pages, so I'd appreciate any help from those who can read them.

### CODE REVIEW

File: src/nvim/main.c [hunk 1]
```diff
@@ -1159,10 +1096,6 @@ static void command_line_scan(mparm_T *parmp)
           }
           break;
 
-        case 'v':                 /* "-v"  Vi-mode (as if called "vi") */
-          exmode_active = 0;
```
Comment (Pyrohh) [inline src/nvim/main.c]: This probably wasn't used much, so I didn't make it a no-op. What do you think @elmart?

File: src/nvim/main.c [hunk 2]
```diff
@@ -205,11 +205,9 @@ int main(int argc, char **argv)
...
   check_and_set_isatty(&params);
 
-  /*
-   * Figure out the way to work from the command name argv[0].
-   * "view" sets "readonlymode", "rvim" sets "restricted", etc.
-   */
-  parse_command_name(&params);
+  // Get the name with which Nvim was invoked, with and without path.
+  set_vim_var_string(VV_PROGPATH, (char_u *)argv[0], -1);
+  set_vim_var_string(VV_PROGNAME, path_tail((char_u *)argv[0]), -1);
```
Comment (Pyrohh) [inline src/nvim/main.c]: These were formerly part of `parse_command_name()`.

File: src/nvim/main.c (hunk 1 above)
Comment (justinmk) [inline src/nvim/main.c]: :+1:  It should not be a noop because the result would be broken.

**Comment by Pyrohh** [comment]
I rebased and made some minor changes for consistency, so please review when you get a chance. The line count shouldn't be too intimidating, as the vast majority of this PR is just documentation.

**Comment by Pyrohh** [comment]
rebased again

**Comment by Pyrohh** [comment]
@justinmk @tarruda could you review this when you get a chance? I'm sorry to keep pinging you specifically @justinmk, I'm just not sure who else should be reviewing this. @elmart maybe? I don't want to be a burden given how much time you already put into this project, hence me trying to balance it out.

File: runtime/doc/starting.txt [hunk 3]
```diff
@@ -48,22 +48,18 @@ filename	One or more file names.  The first one will be the current
...
 		that is read from stdin.  The commands that would normally be
 		read from stdin will now be read from stderr.  Example: >
-			find . -name "*.c" -print | vim -
+			find . -name "*.c" -print | nvim -
 <		The buffer will be marked modified, because it contains text
 		that needs to be saved.  Except when in readonly mode, then
 		the buffer is not marked modified.  Example: >
-			ls | view -
-<
+			ls | nvim -R -
```
Comment (justinmk) [inline runtime/doc/starting.txt]: why remove the `-<` ?

File: runtime/doc/starting.txt [hunk 4]
```diff
@@ -245,28 +215,20 @@ argument.
...
 
 							*-s-ex*
--s		Silent or batch mode.  Only when Vim was started as "ex" or
-		when preceded with the "-e" argument.  Otherwise see |-s|,
-		which does take an argument while this use of "-s" doesn't.
-		To be used when Vim is used to execute Ex commands from a file
-		instead of a terminal.  Switches off most prompts and
-		informative messages.  Also warnings and error messages.
-		The output of these commands is displayed (to stdout):
+-s		Silent or batch mode.  Only when "-s" is preceded by the "-e"
```
Comment (justinmk) [inline runtime/doc/starting.txt]: Wow, the old description was hard to comprehend :)

**Comment by justinmk** [comment]
Assuming the man pages are correct, LGTM. nice choices, and thorough :+1:

File: runtime/doc/starting.txt (hunk 4 above)
Comment (Pyrohh) [inline runtime/doc/starting.txt]: Good riddance :smile:

**Comment by Pyrohh** [comment]
Thanks for reviewing, by the way when you said
```
why remove the < ?
```
was that not an actual issue? I'll add it to my "misc. docs fixes/updates" PR if it's an actual issue.

**Comment by justinmk** [comment]
The syntax highlighting should still look correct (a "convenience" of help file syntax), but I would prefer not to delete the closing `<`.

**Pull request merged**